locator/token_metadata.cc

```diff
@@ -1355,24 +1349,22 @@ void token_metadata_impl::set_pending_ranges(const sstring& keyspace_name,
         _pending_ranges_interval_map[keyspace_name] +=
```

Comment:
As a follow-up, we don't need to lookup `keyspace_name` in `_pending_ranges_interval_map` for every entry.
We can build the keyspace interval map and when done move it to `_pending_ranges_interval_map[keyspace_name]`